app/editor/src/features/content/validation/ContentFormSchema.ts

```diff
@@ -4,42 +4,42 @@ import { array, date, number, object, string } from 'yup';
 export const ContentFormSchema = object().shape(
   {
     productId: number().required('Product is required.').notOneOf([0], 'Product is required.'),
-    sourceId: number().when('tempSource', (value: any) => {
-      if (value === undefined)
+    sourceId: number().when('tempSource', (value: (number | undefined)[]) => {
```

Comment:
@CelloLiu I haven't confirmed all this but you need to update all the schema's to correctly validate with the new yup version.